Sources/Linq2DynamoDb.DataContext/EntityWrapper.cs

```diff
@@ -14,6 +16,31 @@ internal class EntityWrapper : IEntityWrapper
         private readonly IEntityKeyGetter _keyGetter;
         private Document _doc, _newDoc;
 
+        private PropertyInfo _entityVersionNumberProperty;
+        private bool _hasResolvedEntityVersionNumberProperty;
+
+        /// <summary>
+        /// Gets PropertyInfo for the entity's property that has [DynamoDBVersion] 
+        /// attribute or returns null if there is none.
+        /// </summary>
+        private PropertyInfo EntityVersionNumberProperty {
+            get {
+                if (!_hasResolvedEntityVersionNumberProperty) {
+                    _entityVersionNumberProperty = Entity
+                        .GetType()
+                        .GetProperties(BindingFlags.Public | BindingFlags.Instance)
+                        .Where(property =>
+                            property
+                                .GetCustomAttributes(typeof(DynamoDBVersionAttribute), true)
+                                .SingleOrDefault() != null
```

Comment:
or maybe you WANT to throw an exception if an entity contains more than one Version property.
is that even legal in dynamo's object model?

either way.... if your intention is to validate Single versioned property, maybe that should be done somewhere else or at least throw a useful exception if entities violate that constraint.